test/scheduler_patch_test.rb

```diff
@@ -0,0 +1,49 @@
+# vim:fileencoding=utf-8
+require_relative 'test_helper'
+require 'resque/scheduler_patch'
+
+context 'Resque::SchedulerPatch' do
+  setup do
+    Resque::Scheduler.clear_schedule!
+    Resque::Scheduler.configure do |c|
+      c.dynamic = false
+      c.quiet = true
+      c.poll_sleep_amount = nil
+    end
+
+    @scheduler = Resque::Scheduler.dup
+    @scheduler.singleton_class.prepend(Resque::SchedulerPatch)
+  end
+
+  teardown do
+    Resque.redis.redis.flushall
+  end
+
+  test 'can enqueue into new queue format' do
+    t = Time.now + 60 # in the future
+    Resque.enqueue_at(t, SomeIvarJob)
+    Resque.enqueue_at(t, SomeIvarJob)
+    @scheduler.handle_delayed_items(t)
+
+    assert_equal 0, Resque.delayed_queue_schedule_size
+  end
+
+  test 'reads from old format before new one' do
+    t = Time.now + 60 # in the future
+    @scheduler.old_resque.enqueue_at(t, SomeIvarJob)
+    @scheduler.old_resque.enqueue_at(t, SomeIvarJob)
+
+    assert_equal 1, Resque.redis.zcard(:delayed_queue_schedule)
+
+    Resque.enqueue_at(t, SomeIvarJob)
+    Resque.enqueue_at(t, SomeIvarJob)
+
+    assert_equal 2, Resque.delayed_queue_schedule_size
+
+    # Check that next job is from old first
+    @scheduler.handle_delayed_items(t)
+
+    assert_equal 0, Resque.delayed_queue_schedule_size
+    assert_equal 0, Resque.redis.zcard(:delayed_queue_schedule)
+  end
```

Comment:
Can you assert that parameters are untouched and actually test the order that we pop the jobs in? (with an array like assertion)